app/src/main/kotlin/com/wire/android/ui/calling/SharedCallingViewModel.kt

```diff
@@ -0,0 +1,150 @@
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.id.parseIntoQualifiedID
+import com.wire.kalium.logic.feature.call.usecase.EndCallUseCase
+import com.wire.kalium.logic.feature.call.usecase.GetAllCallsUseCase
+import com.wire.kalium.logic.feature.call.usecase.MuteCallUseCase
+import com.wire.kalium.logic.feature.call.usecase.SetVideoPreviewUseCase
+import com.wire.kalium.logic.feature.call.usecase.UnMuteCallUseCase
+import com.wire.kalium.logic.feature.call.usecase.UpdateVideoStateUseCase
+import com.wire.kalium.logic.feature.conversation.ObserveConversationDetailsUseCase
+import com.wire.kalium.logic.util.PlatformView
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@Suppress("LongParameterList")
+@HiltViewModel
+class SharedCallingViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val navigationManager: NavigationManager,
+    private val conversationDetails: ObserveConversationDetailsUseCase,
+    private val allCalls: GetAllCallsUseCase,
+    private val endCall: EndCallUseCase,
+    private val muteCall: MuteCallUseCase,
+    private val unMuteCall: UnMuteCallUseCase,
+    private val updateVideoState: UpdateVideoStateUseCase,
+    private val setVideoPreview: SetVideoPreviewUseCase,
+    private val callRinger: CallRinger
+) : ViewModel() {
+
+    var callState by mutableStateOf(CallState())
+
+    val conversationId: QualifiedID = savedStateHandle
+        .get<String>(EXTRA_CONVERSATION_ID)!!
+        .parseIntoQualifiedID()
+
+    init {
+        viewModelScope.launch {
+            launch {
+                initializeScreenState()
+            }
+            launch {
+                initializeCallingButtons()
+            }
+        }
+    }
+
+    private suspend fun initializeScreenState() {
+        conversationDetails(conversationId = conversationId)
+            .collect { details ->
+                callState = when (details) {
+                    is ConversationDetails.Group -> {
+                        callState.copy(
+                            conversationName = getConversationName(details.conversation.name),
+                            conversationType = ConversationType.Conference
+                        )
+                    }
+                    is ConversationDetails.OneOne -> {
+                        callState.copy(
+                            conversationName = getConversationName(details.otherUser.name),
+                            avatarAssetId = details.otherUser.completePicture?.let { assetId -> ImageAsset.UserAvatarAsset(assetId) },
+                            conversationType = ConversationType.OneOnOne
+                        )
+                    }
+                    is ConversationDetails.Self -> throw IllegalStateException("Invalid conversation type")
+                }
+            }
+    }
+
+    private suspend fun initializeCallingButtons() {
```

Comment:
minor thing but, how about renaming it to initializeCallSettings or something so that we do not suggest here some UI stuff

Reply:
Those buttons are not settings ones